### PR TITLE
Add support for EBS volume encryption

### PR DESCRIPTION
**Pull request by moadqassem**
**What this PR does / why we need it**:
This indicates whether the encryption state of an EBS volume is changed while being restored from a backing snapshot. 

**Which issue(s) this PR fixes** 
Fixes #

```release-note
None
```

### CODE REVIEW

File: pkg/cloudprovider/provider/aws/provider.go [hunk 1]
```diff
@@ -509,6 +514,7 @@ func (p *provider) Create(machine *v1alpha1.Machine, data *cloudprovidertypes.Pr
 					DeleteOnTermination: aws.Bool(true),
 					VolumeType:          aws.String(config.DiskType),
 					Iops:                config.DiskIops,
+					Encrypted:           pointer.BoolPtr(config.EBSVolumeEncrypted),
```
Comment (alvaroaleman) [inline pkg/cloudprovider/provider/aws/provider.go]: Can we add an e2e test where this is enabled?

File: test/e2e/provisioning/testdata/machinedeployment-aws.yaml [hunk 2]
```diff
@@ -33,6 +33,7 @@ spec:
             instanceProfile: "kubernetes-v1"
             diskSize: 50
             diskType: "gp2"
+            ebsVolumeEncrypted: false
```
Comment (alvaroaleman) [inline test/e2e/provisioning/testdata/machinedeployment-aws.yaml]: Please also add this to the sample manifest in `examples/`

File: pkg/cloudprovider/provider/aws/provider.go (hunk 1 above)
Comment (moadqassem) [reply pkg/cloudprovider/provider/aws/provider.go]: I ran the current e2e tests with this sat to `true` and the tests are passed( I am not sure though if this change has any influence on the machine, e.g: the machine will not be provisioned). We can add another test with this enabled. WDYT?

Comment (alvaroaleman) [reply pkg/cloudprovider/provider/aws/provider.go]: sgtm, but I don't think we need to test this on all kube versions and all distributions, one version on one distribution should be enough

**Comment by moadqassem** [comment]
@alvaroaleman PTAL

**Comment by alvaroaleman** [comment]
> @alvaroaleman PTAL

So where is the green e2e run as requested?

**Comment by moadqassem** [comment]
> > @alvaroaleman PTAL
> 
> So where is the green e2e run as requested?

https://public-prow.loodse.com/view/gcs/prow-public-data/pr-logs/pull/kubermatic_machine-controller/663/pull-machine-controller-e2e-aws-ebs-encryption-enabled/1207254364134051840

File: .prow.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+presubmits:
+  #########################################################
+  # unit tests
```
Comment (alvaroaleman) [inline .prow.yaml]: Also no unittest. Other than that, looks good.

**Pull request merged**